### ISSUE TITLE
Feature request: a way to determine "loading" or "rendering" state in noResultsRow (VtClientTable)

### ISSUE THREAD

**Issue by cmitz** [matfish2/vue-tables-2#884]
I would like a better way to "know" when there's actually no data in the overridden `noResultsRow` component (versus data being parsed/rendered).

Options might be:
- passing more data to props
  - Pass allFilteredData 
  - Pass something like allTableDataEmpty
- utilize the `loading` field of `props` to be initially true (e.g. while still rendering)

### Context

We have been building a page based on VtClienTable. We use Vuex-ORM to get state, and we have a lot of rows. Normally around 300, but sometimes more than 3000. We override the `noResultsRow` so we can indicate that we're still loading all required data: we display some "skeleton" rows (grey placeholders for data).

Because the noResultsRow component gets very little information via props, we (mis)use Vuex by just looking up if there's any data in the store.

Showing "No records" while still loading would be confusing for the customer.

### Problem

The `data` prop is a computed property that goes from `[]` to `[...]` with 3000 rows. Immediately after that the page becomes unresponsive (not strange, it's a lot of data), but the `noResultsRow` is visible.

<sub>Image of our table when API call is finished, but VtClientTable is still rendering/parsing/?<sub>
![image](https://user-images.githubusercontent.com/5594436/93994621-62a01e00-fd90-11ea-8c59-fe819f292383.png)

**Comment by matfish2** [comment]
If I understand correctly you are referring to two separate issues:

1. You need a way to "tell" the client table that you are in a loading state rather than just passing an empty dataset.
2. When your API finally returns the data and you assign it to your data property, the table does not respond.

I can solve 1 by adding a setLoadingState(true/false) method to the client table.

As for problem 2 I was unable to reproduce. See [here](https://jsfiddle.net/23pnrgza/) where I simulate an API call by using `setTimeOut`

**Comment by cmitz** [comment]
I think you're right, solution one would be neat.

I will look into your example about the second one, will report how that goes next week :)

**Comment by matfish2** [comment]
https://matanya.gitbook.io/vue-tables-2/client-table/asynchronous-loading

**Comment by matfish2** [comment]
Just tested with 3000 rows and loading wasn't delayed. That being said, your data objects might be more complicated (as you say they are nested 2 levels). I would suggest trying two things: 

1. Strip away any unnecessary complexity. I.e remove data that you don't need or reduce nesting and pass the result to the table.
2. If that still doesn't do the trick, you could add the data in chunks, i.e iterate over the dataset and add, say 100 rows at every iteration.

**Comment by cmitz** [comment]
I won't be creating an example after all (like I mentioned) because the tips helped a lot. Also, I have the feeling the table was already more responsive anyway with the latest version.

We've made some improvements to complexity, and some other changes. Earlier we passed a huge dataset and filtered out most of it via `initFilters`, but now we do the filtering in Vuex ORM instead 👍 (should've done that sooner). It's asynchronous now 🚀 

The `setLoadingState` is a pretty sweet addition too! Thanks!
